Algorithm/QCAlgorithm.Indicators.cs

```diff
@@ -417,13 +417,30 @@ public LinearWeightedMovingAverage LWMA(string symbol, int period, Resolution? r
         public OnBalanceVolume OBV(string symbol, Resolution? resolution = null,
             Func<BaseData, TradeBar> selector = null)
         {
-            string name = CreateIndicatorName(symbol, "OBV", resolution);
+            var name = CreateIndicatorName(symbol, "OBV", resolution);
             var onBalanceVolume = new OnBalanceVolume(name);
-            RegisterIndicator(symbol, onBalanceVolume, resolution);
+            RegisterIndicator(symbol, onBalanceVolume, resolution, selector);
             return onBalanceVolume;
         }
 
         /// <summary>
+        /// Creates a new Average Directional Index indicator. 
+        /// The indicator will be automatically updated on the given resolution.
+        /// </summary>
+        /// <param name="symbol">The symbol whose Average Directional Index Average Directional Index we seek</param>
+        /// <param name="resolution">The resolution.</param> 
+        /// <param name="period">The period over which to compute the Average Directional Index</param>
+        /// <param name="selector">Selects a value from the BaseData to send into the indicator, if null defaults to the Value property of BaseData (x => x.Value).</param>
```

Comment:
...'if null defaults to ( x => (TraderBar)x)'

Since the ADX accepts trade bar instances we don't default the selector to the .Value property of BaseData, but instead use a caste to TradeBar, the assumption here is that if you're using the ADX then you're receiving trade bar data